TACC/Core-Portal
# Task/FP-372 - Add Longhorn and support for multiple storage systems

## Overview: ##

## PR Status: ##


## Related Jira tickets: ##

* [FP-372](https://jira.tacc.utexas.edu/browse/FP-372)

## Summary of Changes: ##
- Updated DataFiles to support multiple storage systems in UI
- Created new settings dictionaries for storage systems and execution systems
- Updated onboarding process for multiple storage systems for a user
- Consolidated UserHome, UserWorkHome, UserAbstractHome managers into UserSystemsManager

## Testing Steps: ##
1. Update `settings_secret.py` with the following:
```
_PORTAL_DATA_DEPOT_LOCAL_STORAGE_SYSTEM_DEFAULT = 'frontera'
_PORTAL_DATA_DEPOT_LOCAL_STORAGE_SYSTEMS = {
    'frontera': {
        'name': 'My Data (Frontera)',
        'prefix': 'frontera.home.{}',
        'host': 'frontera.tacc.utexas.edu',
        'home_dir': '/home1',
        'storage_port': 22,
        'icon': None,
    },
    'longhorn': {
        'name': 'My Data (Longhorn)',
        'prefix': 'longhorn.home.{}',
        'host': 'longhorn.tacc.utexas.edu',
        'home_dir': '/home',
        'storage_port': 22,
        'requires_allocation': 'longhorn3',
        'icon': None,
    },
}
```
2. Log out, then log back in. This will kick off the task to generate your longhorn system.
3. Navigate to Data Files. If you have a longhorn allocation (active or not) you should see the longhorn system available.
4. If you want to test the onboarding process from scratch, close your session and shell into the django container. Delete your user profile:
```
from django.contrib.auth.models import User
from portal.apps.accounts.models import PortalProfile
user = User.objects.get(username='yourusername')
profile = PortalProfile.objects.get(user=user)
profile.delete()
user.delete()
```
And delete your keys:
```
user.ssh_keys.delete()
```
And delete your storage systems:
```
systems-delete frontera.home.${username}
```

5. log into the portal from an incognito window. You should be able to to access the My Data area
6. try the push keys modal for both storage systems

## UI Photos:
![Screen Shot 2020-08-05 at 6 22 20 PM](https://user-images.githubusercontent.com/29575979/89473511-a4292b00-d748-11ea-9123-a819543ae41a.png)


## Notes: ##
